pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -546,7 +583,11 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 
 	// Run the pre-stop lifecycle hooks if applicable.
 	if containerSpec.Lifecycle != nil && containerSpec.Lifecycle.PreStop != nil {
-		gracePeriod = gracePeriod - m.executePreStopHook(pod, containerID, containerSpec, gracePeriod)
+                if containerSpec.Lifecycle.PreStop.IsRetryUntilSuccess {
+			gracePeriod = gracePeriod - m.executeRetryUntilSuccessHook(pod, containerID, containerSpec, gracePeriod)
```

Comment:
Don't worry, the retry time is greater than graceperiod this will not happen because the graceperiod will trigger case <-time.After(time.Duration(gracePeriod) * time.Second).so will return。after this  i will write wait.Until replace this code.